### PR TITLE
docs: Add Snyk status badge to README.md

### PR DESCRIPTION
**Pull request by terrytangyuan**
This is to draw attention to security issues. Snyk has been failing for a couple of days but we haven't noticed even before recent patch releases (example fix for a recent issue https://github.com/argoproj/argo-workflows/pull/12515 but the patch release is already out before this). https://github.com/argoproj/argo-workflows/actions/workflows/snyk.yml?query=branch%3Amain

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 <!-- markdownlint-disable-next-line MD041 -->
 [![slack](https://img.shields.io/badge/slack-argoproj-brightgreen.svg?logo=slack)](https://argoproj.github.io/community/join-slack)
+[![CI Status](https://github.com/argoproj/argo-workflows/workflows/CI/badge.svg)](https://github.com/argoproj/argo-workflows/actions?query=event%3Apush+branch%3Amain)
```
Comment (agilgur5) [inline README.md:3]: CI usually fails though as it is often cancelled and has false positives; you had removed it in #11895

**Comment by terrytangyuan** [comment]
Unrelated: license scan has been failing. @agilgur5 Could you take a look?

**Review by agilgur5 — COMMENTED** [review]
I asked in https://github.com/argoproj/argo-workflows/pull/12470#issuecomment-1879760019, but is Dependabot just behind Snyk? Since it should auto-update on security vulns.

Although in the case of #12515, there is no way to auto-update since it was using the Argo fork of `go-git`

Also left 1 comment in-line

**Comment by agilgur5** [comment]
> Unrelated: license scan has been failing for a long time. @agilgur5 Could you take a look?

This is waiting on Crenshaw ([earlier Slack link](https://cloud-native.slack.com/archives/C020XM04CUW/p1700513395667059?thread_ts=1697816118.936309&cid=C020XM04CUW)) or someone else with existing FOSSA permissions to the Argoproj org, which I don't have

**Comment by terrytangyuan** [comment]
> > Unrelated: license scan has been failing for a long time. @agilgur5 Could you take a look?
> 
> This is waiting on Crenshaw ([earlier Slack link](https://cloud-native.slack.com/archives/C020XM04CUW/p1700513395667059?thread_ts=1697816118.936309&cid=C020XM04CUW)) or someone else with existing FOSSA permissions to the Argoproj org, which I don't have

Can I remove the badge for now?

**Comment by terrytangyuan** [comment]
Yeah dependabot is not fast enough.

**Comment by terrytangyuan** [comment]
Removed CI badge

**Comment by agilgur5** [comment]
> Can I remove the badge for now?

It impacts our CLOMonitor score per #12032. So it would remove one failing badge in favor of a lower score in another badge, which I don't think is ideal either 😕 . I'd much rather fix the root cause (though I don't have permissions to do that myself)

File: README.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 <!-- markdownlint-disable-next-line MD041 -->
 [![slack](https://img.shields.io/badge/slack-argoproj-brightgreen.svg?logo=slack)](https://argoproj.github.io/community/join-slack)
+[![Security Status](https://github.com/argoproj/argo-workflows/workflows/Snyk/badge.svg)](https://github.com/argoproj/argo-workflows/actions/workflows/snyk.yml?query=branch%3Amain)
```
Comment (agilgur5) [inline README.md:3]:
```suggestion
[![Snyk Status](https://github.com/argoproj/argo-workflows/workflows/Snyk/badge.svg)](https://github.com/argoproj/argo-workflows/actions/workflows/snyk.yml?query=branch%3Amain)
```

that would match the existing alt text, but nbd

**Review by agilgur5 — APPROVED** [review]

**Comment by terrytangyuan** [comment]
> would trade one failing badge in favor of a lower score 
How much lower?

**Comment by agilgur5** [comment]
> How much lower?

I don't know, it doesn't explicitly say in the docs (and I don't remember what it was beforehand), so I'd have to check its code to verify

**Comment by agilgur5** [comment]
@terrytangyuan I got the license scan to pass now, see details in https://github.com/argoproj/argo-workflows/pull/12023#issuecomment-1891102433 and further comments.
It's still not quite building daily / latest `main` and still needs transferring though; followed up in [the Slack thread](https://cloud-native.slack.com/archives/C020XM04CUW/p1705271295909369?thread_ts=1697816118.936309&cid=C020XM04CUW)

**Comment by terrytangyuan** [comment]
Great to hear! Thanks!

**Pull request merged**